.github/workflows/resource-cleanup.yml

```diff
@@ -53,3 +53,76 @@ jobs:
         run: |
           python -m pip install -r requirements.txt
           python cleaner.py
+
+  cleanup-k8s-on-ec2:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        instance: [
+          { repo_name: 'amazon-cloudwatch-agent-operator', ec2_name: 'cw-agent-operator-release', language: 'java' },
+          { repo_name: 'amazon-cloudwatch-agent-operator', ec2_name: 'cw-agent-operator-release', language: 'python' },
+          { repo_name: 'amazon-cloudwatch-agent', ec2_name: 'cw-agent-release', language: 'java' },
+          { repo_name: 'amazon-cloudwatch-agent', ec2_name: 'cw-agent-release', language: 'python' },
+          { repo_name: 'aws-otel-python-instrumentation', ec2_name: 'adot-python-release', language: 'python' },
+          { repo_name: 'aws-otel-java-instrumentation', ec2_name: 'adot-java-release', language: 'java' },
+          { repo_name: 'aws-application-signals-test-framework', ec2_name: 'python-canary', language: 'python' },
+          { repo_name: 'aws-application-signals-test-framework', ec2_name: 'java-canary', language: 'java' } ]
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Check if today is the first day of the month
+        id: check_day
+        run: |
+          echo "is_first_day=$(( $(date +%d) == 01 ))" >> $GITHUB_OUTPUT
```

Comment:
We only run all this logic once a month? Would seem safer to do a full check/cleanup every day and detect unneeded clusters

Does this necessarily run after the replace jobs? What happens if this fails or if the replacement jobs fail and need to be rerun? I think we should have this always run